src/Extensions/HttpContextExtensions.cs

```diff
@@ -27,7 +27,7 @@ public static Device GetDevice(this HttpContext context)
                 throw new ArgumentNullException(nameof(context.Items));
 
             return context.Items.TryGetValue(ResponsiveContextKey, out var responsive)
-                       ? (Device) responsive
+                       ? (Device) (responsive ?? Device.Unknown)
```

Comment:
Interesting! `responsive` should never be null at the first place. What test case make you come to this conclusion?